installation fails on R versions < 3.5.0

Using linux (tried multiple flavors including Ubuntu and Amazon Linux), I am trying to install purrr from CRAN.

The error is:

```
trying URL 'https://cloud.r-project.org/src/contrib/purrr_1.0.1.tar.gz'
Content type 'application/x-gzip' length 220628 bytes (215 KB)
==================================================
downloaded 215 KB

* installing *source* package ‘purrr’ ...
** package ‘purrr’ successfully unpacked and MD5 sums checked
** libs
gcc -I/home/btyner/R344/lib64/R/include -DNDEBUG  -I"/home/btyner/R/x86_64-pc-linux-gnu-library/3.4/cli/include" -I/usr/local/include  -fvisibility=hidden -fpic  -g -O0 -mtune=native -c backports.c -o backports.o
gcc -I/home/btyner/R344/lib64/R/include -DNDEBUG  -I"/home/btyner/R/x86_64-pc-linux-gnu-library/3.4/cli/include" -I/usr/local/include  -fvisibility=hidden -fpic  -g -O0 -mtune=native -c cleancall.c -o cleancall.o
gcc -I/home/btyner/R344/lib64/R/include -DNDEBUG  -I"/home/btyner/R/x86_64-pc-linux-gnu-library/3.4/cli/include" -I/usr/local/include  -fvisibility=hidden -fpic  -g -O0 -mtune=native -c coerce.c -o coerce.o
gcc -I/home/btyner/R344/lib64/R/include -DNDEBUG  -I"/home/btyner/R/x86_64-pc-linux-gnu-library/3.4/cli/include" -I/usr/local/include  -fvisibility=hidden -fpic  -g -O0 -mtune=native -c conditions.c -o conditions.o
gcc -I/home/btyner/R344/lib64/R/include -DNDEBUG  -I"/home/btyner/R/x86_64-pc-linux-gnu-library/3.4/cli/include" -I/usr/local/include  -fvisibility=hidden -fpic  -g -O0 -mtune=native -c flatten.c -o flatten.o
gcc -I/home/btyner/R344/lib64/R/include -DNDEBUG  -I"/home/btyner/R/x86_64-pc-linux-gnu-library/3.4/cli/include" -I/usr/local/include  -fvisibility=hidden -fpic  -g -O0 -mtune=native -c init.c -o init.o
gcc -I/home/btyner/R344/lib64/R/include -DNDEBUG  -I"/home/btyner/R/x86_64-pc-linux-gnu-library/3.4/cli/include" -I/usr/local/include  -fvisibility=hidden -fpic  -g -O0 -mtune=native -c map.c -o map.o
map.c: In function ‘map_impl’:
map.c:86:11: warning: implicit declaration of function ‘INTEGER_ELT’; did you mean ‘INTEGER’? [-Wimplicit-function-declaration]
   int n = INTEGER_ELT(ffi_n, 0);
           ^~~~~~~~~~~
           INTEGER
gcc -I/home/btyner/R344/lib64/R/include -DNDEBUG  -I"/home/btyner/R/x86_64-pc-linux-gnu-library/3.4/cli/include" -I/usr/local/include  -fvisibility=hidden -fpic  -g -O0 -mtune=native -c pluck.c -o pluck.o
pluck.c: In function ‘extract_vector’:
pluck.c:130:41: warning: implicit declaration of function ‘COMPLEX_ELT’; did you mean ‘COMPLEX’? [-Wimplicit-function-declaration]
   case CPLXSXP: return Rf_ScalarComplex(COMPLEX_ELT(x, offset));
                                         ^~~~~~~~~~~
                                         COMPLEX
pluck.c:130:41: error: incompatible type for argument 1 of ‘Rf_ScalarComplex’
In file included from pluck.c:3:0:
/home/btyner/R344/lib64/R/include/Rinternals.h:1411:7: note: expected ‘Rcomplex {aka struct <anonymous>}’ but argument is of type ‘int’
 SEXP  Rf_ScalarComplex(Rcomplex);
       ^~~~~~~~~~~~~~~~
/home/btyner/R344/lib64/R/etc/Makeconf:159: recipe for target 'pluck.o' failed
make: *** [pluck.o] Error 1
ERROR: compilation failed for package ‘purrr’
```
It works find in versions of R >= 3.5.0.

--- a/DESCRIPTION
+++ b/DESCRIPTION
@@ -13,7 +13,7 @@ License: MIT + file LICENSE
 URL: https://purrr.tidyverse.org/, https://github.com/tidyverse/purrr
 BugReports: https://github.com/tidyverse/purrr/issues
 Depends: 
-    R (>= 3.4.0)
+    R (>= 3.5.0)
 Imports:
     cli (>= 3.4.0),
     lifecycle (>= 1.0.3),

